src/js/settings/components/feature-additional-settings/nlu-feature.js

```diff
@@ -49,42 +51,37 @@ export const NLUFeatureSettings = () => {
 		},
 	};
 
+	const optionsObjects = {};
+	Object.keys( taxonomies ).forEach( ( postType ) => {
+		if ( featureSettings.post_types?.[ postType ] === postType ) {
+			const postTypeTaxonomies = taxonomies[ postType ] || {};
+			Object.keys( postTypeTaxonomies ).forEach( ( taxonomy ) => {
+				optionsObjects[ taxonomy ] = postTypeTaxonomies[ taxonomy ];
+			} );
+		}
+	} );
+
+	/*
+	 * Add NLU-specific taxonomies to the list if IBM Watson NLU is selected as the provider.
+	 *
+	 * This ensures that the NLU taxonomies are available in the settings,
+	 * as NLU-specific taxonomies are registered only if the Classification feature is enabled and IBM Watson NLU is selected as the provider.
+	 */
+	if ( 'ibm_watson_nlu' === featureSettings.provider ) {
+		Object.keys( nluTaxonomies ).forEach( ( taxonomy ) => {
+			optionsObjects[ taxonomy ] = nluTaxonomies[ taxonomy ];
+		} );
+	}
```

Comment:
Since IBM Watson NLU-specific taxonomies are registered only when the Classification feature is **enabled** and IBM Watson NLU is selected as the provider (saved in the database), and the page is no longer being reloaded, this ensures that those taxonomies are available in the settings. Open to discussing better alternative approaches to this.